airbyte-ci/connectors/metadata_service/lib/tests/test_gcs_upload.py

```diff
@@ -155,3 +156,41 @@ def test_upload_metadata_to_gcs_invalid_docker_images(mocker, invalid_metadata_u
                 "my_bucket",
                 metadata_file_path,
             )
+
+
+def test_upload_metadata_to_gcs_with_prerelease(mocker, valid_metadata_upload_files):
+    # Arrange
+    setup_upload_mocks(mocker, "new_md5_hash1", "new_md5_hash2", "new_md5_hash3")
+    mocker.patch("metadata_service.gcs_upload._latest_upload", return_value=None)
+    mocker.patch("metadata_service.gcs_upload.upload_file_if_changed", return_value=None)
+    mocker.spy(gcs_upload, "_prerelease_upload")
+
+    for valid_metadata_upload_file in valid_metadata_upload_files:
+        # Assuming there is a valid metadata file in the list, if not, you might need to create one
+        metadata_file_path = pathlib.Path(valid_metadata_upload_file)
+        prerelease_image_tag = "prerelease_image_tag"
+
+        gcs_upload.upload_metadata_to_gcs(
+            "my_bucket",
+            metadata_file_path,
+            prerelease_image_tag,
+        )
+
+        gcs_upload._latest_upload.assert_not_called()
+
+        # Assert that _prerelease_upload is called and the third argument is prerelease_image_tag
+        # Ignore the first and second arguments (_ and __ are often used as placeholder variable names in Python when you don't care about the value)
+        _, __, prerelease_tag = gcs_upload._prerelease_upload.call_args[0]
+        assert prerelease_tag == prerelease_image_tag
+
+        # Verify the tmp metadata is overridden
+        metadata, error = gcs_upload.validate_and_load(metadata_file_path, [])
+        expected_tmp_metadata_file_path = pathlib.Path("/tmp") / metadata.data.dockerRepository / prerelease_image_tag / METADATA_FILE_NAME
+        assert expected_tmp_metadata_file_path.exists(), f"{expected_tmp_metadata_file_path} does not exist"
+
+        # verify that the metadata is overrode
+        tmp_metadata, error = gcs_upload.validate_and_load(expected_tmp_metadata_file_path, [])
+        tmp_metadata_dict = to_json_sanitized_dict(tmp_metadata, exclude_none=True)
+        for registry in get(tmp_metadata_dict, "data.registries", {}).values():
+            if "dockerImageTag" in registry:
+                assert registry["dockerImageTag"] == prerelease_image_tag
```

Comment:
let's validate the main `dockerImageTag` as well since that one is most likely to be used